pybop/_problem.py

```diff
@@ -44,30 +93,67 @@ def __init__(
         if len(self._target) != len(self._time_data):
             raise ValueError("Time data and signal data must be the same length.")
 
-        # Set bounds
-        self.bounds = dict(
-            lower=[param.bounds[0] for param in self.parameters],
-            upper=[param.bounds[1] for param in self.parameters],
+        # Build the model
+        if self._model._built_model is None:
+            self._model.build(
+                dataset=self._dataset,
+                parameters=self.parameters,
+                check_model=self.check_model,
+                init_soc=self.init_soc,
+            )
+
+    def evaluate(self, parameters):
+        """
+        Evaluate the model with the given parameters and return the signal.
+        """
+
+        y = np.asarray(self._model.simulate(inputs=parameters, t_eval=self._time_data))
+
+        return y
+
+    def evaluateS1(self, parameters):
+        """
+        Evaluate the model with the given parameters and return the signal and
+        its derivatives.
+        """
+        for i, key in enumerate(self.parameters):
+            self.parameters[key] = parameters[i]
+
+        y, dy = self._model.simulateS1(
+            inputs=self.parameters,
+            t_eval=self._time_data,
         )
 
-        # Sample from prior for x0
-        if x0 is None:
-            self.x0 = np.zeros(self.n_parameters)
-            for i, param in enumerate(self.parameters):
-                self.x0[i] = param.rvs(1)
-        elif len(x0) != self.n_parameters:
-            raise ValueError("x0 dimensions do not match number of parameters")
+        return (np.asarray(y), np.asarray(dy))
 
-        # Add the initial values to the parameter definitions
-        for i, param in enumerate(self.parameters):
-            param.update(value=self.x0[i])
 
-        # Set the fitting parameters and build the model
-        self.fit_parameters = {o.name: o.value for o in parameters}
-        if self._model._built_model is None:
+class DesignProblem(BaseProblem):
+    """
+    Defines the problem class for a design optimiation problem.
+    """
+
+    def __init__(
+        self,
+        parameters,
+        experiment,
+        model,
+        check_model=True,
+        init_soc=None,
+        x0=None,
+    ):
+        super().__init__(parameters, model, check_model, init_soc, x0)
+        self.experiment = experiment
+        self._target = None
+
+        # Build the model if required
+        if experiment is not None:
+            # Leave the build until later to apply the experiment
+            self._model.parameters = self.parameters
+
+        elif self._model._built_model is None:
             self._model.build(
-                dataset=self._dataset,
-                fit_parameters=self.fit_parameters,
+                experiment=self.experiment,
```

Comment:
Is experiment required in the model.build() method? Maybe for future functionality, perhaps?

Reply:
Good spot - yes maybe but I'll leave it out for now.